### DISCUSSION TITLE
Extendable language detection for spaCy on Doc and Span objects

### DISCUSSION THREAD

**Discussion by Abhijit-2592** [explosion/spaCy#3265] [Show & Tell]
Hi guys,

First of all thanks for the awesome NLP library. #1172 is a feature request for a built in language detection module (this is closed now, so I am opening a new one!). @ines  mentioned that this can be achieved using custom pipelines in spacy>=2.0. And @honnibal mentioned that this is in plan. Since I needed this feature, I wrote a small wrapper [spacy-langdetect](https://github.com/Abhijit-2592/spacy-langdetect). Out of the box this can be extended to use any language detector.

@ines and @honnibal any plans on spaCy's own language detector? Or, is there anyway to fit this into spaCy? Kindly let me know of your thoughts

Once again thanks for this beautiful library!

**Comment by ines** [comment]
Ohhh nice! 👍 I also like that the plugin can be extended with custom language detectors. I think this is a good use case for a plugin, since it depends on other third-party libraries that we wouldn't necessarily want to have as a dependency of the core library. Maybe we'll add language detection to the core library at some point in the future, but it's not a priority and for now, a plugin like that is a great solution.

We should definitely add this to the [spaCy universe](https://spacy.io/universe) btw to make it easier for others to find and use it. If you like, you can submit a PR to the [`universe.json`](https://github.com/explosion/spacy/blob/master/website/universe/universe.json) – otherwise, I'm also happy to add it.

**Comment by Abhijit-2592** [comment]
@ines Thanks! I have created the pull request #3266

**Comment by ines** [comment]
Perfect. This will be added when we next update the website.
I've also just shared it on Twitter: https://twitter.com/spacy_io/status/1095368837011259392

**Comment by Abhijit-2592** [comment]
@ines  Thanks for your quick support!! and thanks for sharing it on twitter
